src/openzaak/components/zaken/tests/test_rol_filter_machtiging.py

```diff
@@ -0,0 +1,405 @@
+            )
+
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+            self.assertEqual(response.json()["count"], 1)
+            self.assertEqual(
+                response.json()["results"][0]["url"],
+                f"http://testserver{reverse(rol_gemachtigde)}",
+            )
+
+        with self.subTest(MachtigingChoices.machtiginggever):
+            response = self.client.get(
+                self.url, {"machtiging": MachtigingChoices.machtiginggever}
+            )
+
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+            self.assertEqual(response.json()["count"], 1)
+            self.assertEqual(
+                response.json()["results"][0]["url"],
+                f"http://testserver{reverse(rol_machtiginggever)}",
+            )
+
+    def test_filter_machtiging_loa(self):
+        zaak = ZaakFactory.create()
+        roltype = RolTypeFactory.create(
+            zaaktype=zaak.zaaktype, omschrijving_generiek=RolOmschrijving.initiator
+        )
+        rol_loa_middle = RolFactory.create(
+            zaak=zaak,
+            roltype=roltype,
+            betrokkene_type=RolTypes.natuurlijk_persoon,
+            indicatie_machtiging=IndicatieMachtiging.gemachtigde,
+            roltoelichting="Created zaak",
+            authenticatie_context={
+                "source": "digid",
+                "level_of_assurance": DigiDLevelOfAssurance.middle,
+                "representee": {"identifier_type": "bsn", "identifier": "111222333"},
+                "mandate": {
+                    "services": [{"id": "5628edbd-333e-460d-8a69-8f083b8cf1b8"}]
+                },
+            },
+        )
+        RolFactory.create(
+            zaak=zaak,
+            roltype=roltype,
+            betrokkene_type=RolTypes.natuurlijk_persoon,
+            indicatie_machtiging=IndicatieMachtiging.gemachtigde,
+            roltoelichting="Created zaak",
+            authenticatie_context={
+                "source": "digid",
+                "level_of_assurance": DigiDLevelOfAssurance.high,
+                "representee": {"identifier_type": "bsn", "identifier": "123456782"},
+                "mandate": {
+                    "services": [{"id": "5628edbd-333e-460d-8a69-8f083b8cf1b8"}]
+                },
+            },
+        )
+
+        with self.subTest(DigiDLevelOfAssurance.middle):
+            response = self.client.get(
+                self.url, {"machtiging__loa": DigiDLevelOfAssurance.middle}
+            )
+
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+            self.assertEqual(response.json()["count"], 1)
+            self.assertEqual(
+                response.json()["results"][0]["url"],
+                f"http://testserver{reverse(rol_loa_middle)}",
+            )
+
+        with self.subTest(DigiDLevelOfAssurance.high):
+            response = self.client.get(
+                self.url, {"machtiging__loa": DigiDLevelOfAssurance.high}
```

Comment:
Yes, this is an expected behavior,  `MaxLoAFilter` is based on `ChoiceFilter`, so if the value in query param is not in the enum, it will return 400
I'll add a test for it!